### ISSUE TITLE
window.location.port is ignored

### ISSUE THREAD

**Issue by masakij** [socketio/socket.io-client#802]
The url parser was ignoring port hints from window.location when the uri string  was relative path.

```javascript
// relative path support
  if ('string' == typeof uri) {
    if ('/' == uri.charAt(0)) {
      if ('/' == uri.charAt(1)) {
        uri = loc.protocol + uri;
      } else {
        uri = loc.hostname + uri; // <- this
      }
    }
```

related issue: #797

**Comment by defunctzombie** [comment]
can you code up a fix and tests?

**Comment by masakij** [comment]
Is this right? #803
(Sorry, I'm not used to using github)

**Comment by adi1133** [comment]
Use location.origin instead of location.hostname, this way the protocol is preserved.

**Comment by masakij** [comment]
location.origin is not compatible with IE 9 & under.

protocol-less url  bind with a protocol on next block.
```javascript
     if (!/^(https?|wss?):\/\//.test(uri)) {
       debug('protocol-less url %s', uri);
       if ('undefined' != typeof loc) {
         uri = loc.protocol + '//' + uri;
       } else {
         uri = 'https://' + uri;
       }
     }
```

**Comment by nnseva** [comment]
I've met the same problem, the #803 should fix the problem, please apply it.

**Comment by masakij** [comment]
What should I do next?
If there is any problem on #803 , I'll fix it.

**Comment by nkzawa** [comment]
Fixed in 1.4.x
